agent/eventhandler/task_handler.go

```diff
@@ -61,6 +66,7 @@ func NewTaskHandler() *TaskHandler {
 		tasksToEvents:          make(map[string]*eventList),
 		submitSemaphore:        utils.NewSemaphore(concurrentEventCalls),
 		tasksToContainerStates: make(map[string][]api.ContainerStateChange),
+		statesaver:             statemanager.NewNoopStateManager(),
```

Comment:
Am I missing something here or do you need an additional change to set `statesaver` to the actual state saver object created in `app/agent.go` ?

Reply:
I'd made an incorrect assumption here which was clarified post discussion.